### PR TITLE
tests: add missing snapshot and axe coverage for ActionMenu

### PR DESCRIPTION
**Pull request by khiga8**
Fixes:  https://github.com/primer/react/issues/4530

### Changelog

I noticed that most of the snapshots test and axe scan coverage for ActionMenu are for the default closed state, which is just a button.

This PR adds an action to open the ActionMenu to ensure we're getting adequate visual regression and axe testing coverage, and regenerates the snapshots. 

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: e2e/components/ActionMenu.test.ts [hunk 1]
```diff
@@ -36,30 +37,29 @@ test.describe('ActionMenu', () => {
       test.describe(theme, () => {
```
Comment (khiga8) [inline e2e/components/ActionMenu.test.ts:37]: I noticed that this test is under `Inactive items` but it's visiting the wrong ID, so updated the ID!

File: .playwright/snapshots/components/ActionList.test.ts-snapshots/ActionList-Block-Description-dark-colorblind-linux.png
Comment (khiga8) [inline .playwright/snapshots/components/ActionList.test.ts-snapshots/ActionList-Block-Description-dark-colorblind-linux.png:1]: The snapshots changed for ActionList bcuz Jonathan's profile picture changed 😅

**Comment by joshblack** [comment]
@khiga8 that's a great point about the CI failing, ideally with `update snapshots` it should go green before merging since it is checking in the correct images. It may be that there is something flakey going on here, will take a look tomorrow if that's okay 👀 Typically this comes up with animations but I think you have `animation: 'disabled'` when taking screenshots so I'm not sure what would be causing it here 🤔

File: e2e/components/ActionMenu.test.ts [hunk 2]
```diff
@@ -46,7 +46,9 @@ test.describe('ActionMenu', () => {
           // Open state
           await page.locator('button', {hasText: 'Open menu'}).waitFor()
           await page.getByRole('button', {name: 'Open menu'}).click()
-          expect(await page.screenshot()).toMatchSnapshot(`ActionMenu.Inactive Items.${theme}.png`)
+          expect(await page.screenshot({animations: 'disabled'})).toMatchSnapshot(
```
Comment (khiga8) [inline e2e/components/ActionMenu.test.ts:49]: Aha! Thank you @joshblack! 🌟

**Comment by joshblack** [comment]
bump @mperrotti when you have a sec to review 👀

**Review by mperrotti — APPROVED** [review]

**Pull request merged**